public/react-services/saved-objects.js

```diff
@@ -42,44 +44,59 @@ export class SavedObject {
    * Returns the full list of index patterns that are valid
    * An index is valid if its fields contain at least these 4 fields: 'timestamp', 'rule.groups', 'agent.id' and 'manager.name'
    */
-  static async getListOfWazuhValidIndexPatterns() {
+  static async getListOfWazuhValidIndexPatterns(defaultIndexPatterns, where) {
     try {
-      const list = await this.getListOfIndexPatterns();
-      const result = list.filter(item => {
-        if (item.attributes && item.attributes.fields) {
-          const fields = JSON.parse(item.attributes.fields);
-          const minimum = {
-            timestamp: true,
-            'rule.groups': true,
-            'manager.name': true,
-            'agent.id': true
-          };
-          let validCount = 0;
-
-          fields.map(currentField => {
-            if (minimum[currentField.name]) {
-              validCount++;
-            }
-          });
+      let result = [];
+      if (where === 'healthcheck') {
+        const list = await Promise.all(
+          defaultIndexPatterns.map(
+            async (pattern) => await SavedObject.getExistingIndexPattern(pattern)
+          )
+        );
+        result = this.validateIndexPatterns(list);
+      }
 
-          if (validCount === 4) {
-            return true;
-          }
-        }
-        return false;
-      });
+      if (!result.length) {
+        const list = await this.getListOfIndexPatterns();
+        result = this.validateIndexPatterns(list);
+      }
 
-      const validIndexPatterns = result.map(item => {
+      return result.map((item) => {
         return { id: item.id, title: item.attributes.title };
       });
-      return validIndexPatterns;
     } catch (error) {
       return ((error || {}).data || {}).message || false
         ? error.data.message
         : error.message || error;
     }
   }
 
+  static validateIndexPatterns(list) {
+    return list.filter(item => {
+      if (item.attributes && item.attributes.fields) {
+        const fields = JSON.parse(item.attributes.fields);
+        const minimum = {
+          timestamp: true,
+          'rule.groups': true,
+          'manager.name': true,
+          'agent.id': true,
+        };
+        let validCount = 0;
+
+        fields.map(currentField => {
+          if (minimum[currentField.name]) {
+            validCount++;
+          }
+        });
+
+        if (validCount === 4) {
+          return true;
+        }
+      }
+      return false;
+    });
+  }
+
```

Comment:
This could be refactored using the `Array.prototype.every` method to check the 4 minimum fields are included in the index pattern fields instead of the old logic. I know it was before in the code.